### DISCUSSION TITLE
Rulesets and Referencing

### DISCUSSION THREAD

**Discussion by ZacharyCJDecker** [earthlab-education/Earth-Analytics-AY24#61] [Q&A] [answered]
When trying to edit my Read Me and submit a pull request I get the error: Cannot create ref due to create name restrictions.

I get the error whether I try to reference the Scavenger Hunt or not. Any ideas?

**Comment by eculler** [comment]
Hi @ZacharyCJDecker  great question! You need to change to the branch that matches your username (ZacharyCJDecker-contrib) in order to make any changes. I'm guessing that GitHub is objecting because you're trying to create a branch something like ZacharyCJDecker-patch-1, which is allowed in some repositories, but not this one (because I want you all to get in the habit of making changes in a branch rather than in main, which is usually considered best practice). To fix you will need to:

1.  Go to the Code Tab
2. Click on the dropdown menu that says `main` in the upper left
3. Select the branch with your username in it
4. Edit the README, making sure your branch is still selected
5. Submit your pull request.

Reply here if you have any further issues!

For others reading this, another reason you might have this problem is that I haven't created your branch yet because I didn't have your GitHub username. If that's the case, go ahead and reply here tagging me (e.g. @eculler) and I will add you.

**Reply by ZacharyCJDecker** [reply]
Hey @eculler 

That makes sense, except that I tried changing the branch and I get the same error. I was thinking I wanted to create a new branch in order to create a pull request. But, if I commit directly to my branch is the pull request still created?

![Error](https://github.com/user-attachments/assets/9d07c53b-9212-4f4c-8b43-4cc8d33e3e57)

**Reply by ZacharyCJDecker** [reply, marked as answer]
We answered this in class. 

I should have chosen the first option to commit directly. Then a second step is required under the Pull Requests page to create the pull request.
